crates/sdk/src/lib.rs

```diff
@@ -835,15 +838,18 @@ pub mod testing {
         ResignSteward(Address),
         PendingTransfer(PendingTransfer),
         IbcAny(Any),
-        Custom(Box<dyn std::fmt::Debug>),
+        Custom,
     }
 
     prop_compose! {
         // Generate an arbitrary commitment
         pub fn arb_commitment()(
-            hash in arb_hash(),
+            commitment in prop_oneof![
+                arb_hash().prop_map(Commitment::Hash),
+                arbitrary::any::<Vec<u8>>().prop_map(Commitment::Id),
```

Comment:
should we emplace an upper bound on the size of these commitments?

Reply:
Yeah this is a possibility, do you have any numbers in mind for the limit/cut-off? But note that the (unrestricted) commitment sizes have not caused practical issues in the test vectors or their use in testing the Ledger app (yet). The Ledger app has been able to handle all the vectors generated so far, even though these vectors are probably being drawn from a superset of what might actually occur in practice.

Reply:
if commitments generated here only represents the happy path, something like 1 KiB at most I think; that said, testing larger commitments would also be great, in case they are too large for the ledger device to handle